src/create/schema/index.ts

```diff
@@ -211,14 +212,45 @@ export const createSchema = <
   return createManualTypeSchema(zodSchema);
 };
 
+export const createSchema = <
+  Output = any,
+  Def extends ZodTypeDef = ZodTypeDef,
+  Input = Output,
+>(
+  zodSchema: ZodType<Output, Def, Input>,
+  state: SchemaState,
+): oas31.SchemaObject | oas31.ReferenceObject => {
+  if (state.visited?.has(zodSchema)) {
+    throw new Error(
+      `The schema at ${
+        state.path?.join(' > ') || '<root>'
+      } needs to be registered because it's circularly referenced`,
+    );
+  }
+  state.visited ??= new Set();
+  state.visited.add(zodSchema);
+  const result = createSchemaSwitch(zodSchema, state);
+  state.visited.delete(zodSchema);
+  return result;
+};
+
 export const createSchemaOrRef = <
   Output = any,
   Def extends ZodTypeDef = ZodTypeDef,
   Input = Output,
 >(
   zodSchema: ZodType<Output, Def, Input>,
   state: SchemaState,
+  subpath?: string[],
```

Comment:
hmmm does this need to be an array?

Reply:
yeah probably not, when I started I was thinking some cases might need to pass multiple elements, but none needed to